docs/source/modules.rst

```diff
@@ -7,31 +7,114 @@ the same pattern as much as possible.
 The API of each module is composed of two parts. The *auth* parameter contains
 the pieces of information that are related to the Sensu Go backend that the
 module is connecting to. All other parameters hold the information related to
-the resource that we are operating on. A typical task would look like this:
+the resource that we are operating on.
+
+
+Authentication parameters
+-------------------------
+
+Each module has an *auth* parameter that holds the following information about
+the Sensu Go backend:
+
+1. The **url** key holds the address of the Sensu Go backend. If this key is
+   not present in the task's definition, Ansible will consult the *SENSU_URL*
+   environment variable and, if the variable is not set, use the default value
+   of ``http://localhost:8080``.
+2. The **user** anf **password** keys contain the credentials that the module
+   will use when connecting to the backend. It not present, Ansible will try
+   to look up the *SENSU_USER* and *SENSU_PASSWORD* environment variables,
+   falling back to the default values of ``admin`` and ``P@ssw0rd!``.
+3. The **api_key** field should contain an `API key`_ that module will use to
+   authenticate with the backend. If not present, Ansible will try to use the
+   value stored in the *SENSU_API_KEY*.
+
+.. _API key:
+   https://docs.sensu.io/sensu-go/latest/guides/use-apikey-feature/
+
+.. note::
+
+   The API key authentication is only available in Sensu Go 5.15 or newer.
+
+When Ansible tries to connect to the Sensu Go backend, it will try to
+authenticate using the API key. If the API key is not set, Ansible will
+fallback to using the *user* and *password* values for authentication. What
+this basically means is that there are two valid sets of *auth* parameters:
 
 .. code-block:: yaml
 
-   - name: Make sure asset is present
-     asset:
+   - name: Use API key authentication
+     sensu.sensu_go.asset:
```

Comment:
Prefixing with namespace & collection name is totally fine, however we now have some examples in this file that use this notation and others that assume `collections: [ sensu.sensu_go ]` is specified in the play. However feel free to ignore this nitpick, since it's true that we demonstrate both ways of referring to collection content in the quickstart document.

Reply:
I will unify the notation. Let us not give our users yet another reason to be confused ;)